Sources/GXUITest/GXUITestingAPI.swift

```diff
@@ -837,12 +837,16 @@ public class SdtUITestSD {
 		if received == expected {
 			return true
 		}
-		else {
+		else if !received.hasPrefix(" AM") || !received.hasPrefix(" PM"){
+			var expectedSpace = received.replacingOccurrences(of: "AM", with: " AM")
+			expectedSpace = received.replacingOccurrences(of: "PM", with: " PM")
+			return received == expectedSpace
+		}
+		else if received.hasPrefix("\n"){
```

Comment:
Several issues here:
1. This code does not compile (not all paths return a value).
2. Condition is not related to prefix but to suffix.

The issue here is iOS 17 replaced AM/PM separator from a SPACE (U+0020) to NARROW NO-BREAK SPACE (U+202F).

The following code would be a better fix, and also does not beak existing "expectedNoNewlines" path.
```suggestion
		else if #available(iOS 17, tvOS 17, watchOS 10, *),
				["\u{202F}AM", "\u{202F}PM"].contains(where: received.hasSuffix(_:)),
				received.count == expected.count {
			/// iOS 17 replaced AM/PM separator from a SPACE (U+0020) to NARROW NO-BREAK SPACE (U+202F)
			let idx = received.lastIndex(of: "\u{202F}")!
			let range = Range<String.Index>(uncheckedBounds: (lower: idx, upper: received.index(after: idx)))
			let expectedWithNarrowNoBreakSpace = expected.replacing("\u{0020}", with: "\u{202F}", subrange: range)
			return received == expectedWithNarrowNoBreakSpace
		}
		else {
```

Reply:
I don't like your suggested approach either.

I think a better approach would be to compare Dates rather than the String representation of dates. The latter are prone to changes because of the device's language settings (i.e. we don't know if Apple made similar changes in other languages where the  "A.M."/"P.M." string are not used).

Reply:
The problem is we don't have access to Dates, but to the string is being displayed.
The developer code VerifyControlValue with a String, and already had to take test device settings into account.
Another option would be to don't do nothing here and require the developer (as already required between platform differences) to deal with the differences in the test code (which in this case is really hard do to being an invisible character which looks like a simple space).

Reply:
Can't we parse both Strings (the control's displayed value and the method input parameter) into Dates?

The semantic of VerifyControlValue is that of a value, not a String per-se (the String parameter type is a GeneXus limitation of not being able to overload a method changing just its parameter types).

Reply:
Initially, converting the strings to Dates and comparing them appears to be working well. I will check the PR that Fabian mentioned in the mail and do the test for the Date approach.

Reply:
How do you know you should try to convert to a Date for comparison?
The only input you have is from verifycontrolvalue(_ controlName: String, _ value: String).
You don't know if its a date you are trying to compare. Which format you use to convert from screen display string?